src/webview/js/generatedEs3.js

```diff
@@ -116,11 +116,14 @@ var isMessageNode = function isMessageNode(node) {
 
 var getStartAndEndNodes = function getStartAndEndNodes() {
   var startNode = getMessageNode(document.elementFromPoint(200, 20));
-  var endNode = getMessageNode(document.elementFromPoint(200, window.innerHeight - 20));
+  var windowInnerHeight = window.innerHeight;
+  var messageListHeight = document.getElementById('message-list').clientHeight;
+  var endPoint = messageListHeight > 0 ? Math.min(windowInnerHeight, messageListHeight) : windowInnerHeight;
+  var endNode = getMessageNode(document.elementFromPoint(200, endPoint - 20));
```

Comment:
This logic isn't right -- in particular, you're counting by a distance `messageListHeight` down from the top of the whole viewport, but the message list may not start at the top of the viewport.

Take a look at [`getBoundingClientRect`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) -- I think it may save you from trying to sort through this by hand, and give shorter and clearer code.